src/main/java/au/com/dmg/fusion/response/diagnosisresponse/DiagnosisResponse.java

```diff
@@ -0,0 +1,119 @@
+    public ResponseExtensionData getResponseExtensionData() { return responseExtensionData; }
+
+    public static class Builder {
+        private Response response;
+        private List<String> loggedSaleID;
+        private POIStatus poiStatus;
+        private List<HostStatus> hostStatus;
+        private ResponseExtensionData responseExtensionData;
+
+        public Builder() {
+        }
+
+        Builder(Response response,List<String> loggedSaleID, POIStatus poiStatus, List<HostStatus> hostStatus, ResponseExtensionData responseExtensionData){
+            this.response = response;
+            this.loggedSaleID = loggedSaleID;
+            this.poiStatus = poiStatus;
+            this.hostStatus = hostStatus;
+            this.responseExtensionData = responseExtensionData;
+        }
+
+        public Builder response(Response response){
+            this.response = response;
+            return Builder.this;
+        }
+        public Builder loggedSaleIDs(List<String> loggedSaleID){
+            this.loggedSaleID = loggedSaleID;
+            return Builder.this;
+        }
+        public Builder loggedSaleID(String loggedSaleID){
+            this.loggedSaleID = Collections.singletonList(loggedSaleID);
+            return Builder.this;
+        }
+        public Builder loggedSaleID(List<String> loggedSaleID){
+            this.loggedSaleID = loggedSaleID;
+            return Builder.this;
+        }
+        public Builder addLoggedSaleID(String loggedSaleID){
+            if(loggedSaleID!=null) {
+                if (this.loggedSaleID != null) {
+                    this.loggedSaleID.add(loggedSaleID);
+                } else {
+                    this.loggedSaleID = Collections.singletonList(loggedSaleID);
+                }
+            }
+                return Builder.this;
+        }
+        public Builder poiStatus(POIStatus poiStatus){
+            this.poiStatus = poiStatus;
+            return Builder.this;
+        }
+        public Builder hostStatus(List<HostStatus> hostStatus){
+            this.hostStatus = hostStatus;
+            return Builder.this;
+        }
+        public Builder hostStatus(HostStatus hostStatus){
+            this.hostStatus = Collections.singletonList(hostStatus);
+            return Builder.this;
+        }
+        public Builder addHostStatus(HostStatus hostStatus){
+            if(hostStatus!=null){
+                if(this.hostStatus!=null){
+                    this.hostStatus.add(hostStatus);
+                }else{
+                    this.hostStatus = Collections.singletonList(hostStatus);
+                }
+            }
+            return Builder.this;
+        }
+
+        public Builder responseExtensionData(ResponseExtensionData responseExtensionData){
+            this.responseExtensionData = responseExtensionData;
+            return Builder.this;
+        }
+
+        public DiagnosisResponse build() {
+            return new DiagnosisResponse(this);
```

Comment:
If we currently use DiagnosisResponse to get POI Information (and Host Status?), then, please add the null checking of ExtensionData (and HostStatus).  Otherwise, Satellite can just respond with an _empty_ DiagnosisResponse without the required data.